scipy/io/tests/test_mmio.py

```diff
@@ -61,6 +73,18 @@ def test_simple_rectangular_integer(self):
         self.check_exact([[1, 2, 3], [4, 5, 6]],
                          (2, 3, 6, 'array', 'integer', 'general'))
 
+    def test_simple_upper_triangle_unsigned_integer(self):
```

Comment:
It may be clearer / more concise to use a [parametrized test](https://docs.pytest.org/en/latest/parametrize.html). For example, this test function is almost identical to the plain integer version. Likewise for a number of the other cases you've added. I'm not sure it is crucial, but probably a bit easier to read / more consistent with pytest style.

Reply:
Yeah I agree it will actually look easier to read. However I didn't want to make it too different from  the current test. 
Just to make sure that we are on the same page
```
@pytest.mark.parametrize("test_input", [[[1, 2], [3, 4]],
                                        [[2, 10], [3, 6]],
                                        [[2, 100], [3, 9]]])
def test_simple_unsigned_integer(self, test_input):
    self.check_exact(array(test_input, 'uint'),
                    (2, 2, 4, 'array', 'unsigned-integer', 'general'))
```
Please let me know if this suits better.

Reply:
I mean reusing the original test instead of creating an almost identical test with a new name just because of additional type support.

Something closer to this, for example:

```python
@pytest.mark.parametrize("typeval", ["integer", "uint"])
def test_simple_integer(self, typeval):
  self.check_exact([[1, 2], [3, 4]],
                    (2, 2, 4, 'array', typeval, 'general'))
```

You'd probably have to add on a dtype parameter too, if I understand correctly, for the mmio field value.

Reply:
Looks like only a small subset of the original tests actually have negative values, so some concision / clarity can still be obtained from parametrization. For the rare cases where negative values are present in the original tests, you could also just parameterize the values or in those specific cases use separate tests.